### PR TITLE
Cleanup FieldInfo tests

### PR DESCRIPTION
**Pull request by hughbe**
/cc @stephentoub

### CODE REVIEW

File: src/System.Reflection/tests/FieldInfoTests.cs [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+        [InlineData(nameof(FieldInfoTests.ConstStringField), "new value")]
+        [InlineData(nameof(FieldInfoTests.ConstCharField), 'A')]
+        [InlineData(nameof(FieldInfoTests.ConstBoolField), false)]
+        [InlineData(nameof(FieldInfoTests.ConstFloatField), 4.56f)]
+        [InlineData(nameof(FieldInfoTests.ConstDoubleField), double.MaxValue)]
+        [InlineData(nameof(FieldInfoTests.ConstInt64Field), long.MaxValue)]
+        [InlineData(nameof(FieldInfoTests.ConstByteField), byte.MaxValue)]
+        public void SetValue_ConstantField_ThrowsFieldAccessException(string field, object value)
+        {
+            FieldInfo fieldInfo = GetField(typeof(FieldInfoTests), field);
```
Comment (stephentoub) [inline src/System.Reflection/tests/FieldInfoTests.cs]: Why did you remove the other asserts?

File: src/System.Reflection/tests/FieldInfoTests.cs [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+            FieldInfo fieldInfo = GetField(type, name);
+            Assert.Equal(expected, fieldInfo.IsFamilyAndAssembly);
+        }
+
+        [Theory]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField1), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField2), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField3), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField4), false)]
+        [InlineData(typeof(FieldInfoTests), nameof(FieldInfoTests.s_assemblyField5), false)]
```
Comment (stephentoub) [inline src/System.Reflection/tests/FieldInfoTests.cs]: I realize you're just cleaning up the ones that were already there.  It'd be good at some point to have an entry that has expected == true.

File: src/System.Reflection/tests/FieldInfoTests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Reflection/tests/FieldInfoTests.cs]: I see; it's not necessary because we wouldn't get back the FieldInfo if the name was different.

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Debug Build and Test please (seg fault in CLI)

@adityamandaleeka, FYI, we've been hitting these seg faults way more frequently than we used to.  I'm a little concerned that, beyond the issue you've already uncovered, we may have made something worse in the runtime that's triggering these more frequently.

**Comment by stephentoub** [comment]
Thanks, @hughbe. LGTM.

File: src/System.Reflection/tests/FieldInfoTests.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Reflection/tests/FieldInfoTests.cs]: I have a todo list of tests to add when I finish the cleanup for System.Reflection, based on this and your other suggestions (keep them up :) ). Just one question: does buildtools/the build infrastructure support IL projects/IL files?

Comment (stephentoub) [inline src/System.Reflection/tests/FieldInfoTests.cs]: > Just one question: does buildtools/the build infrastructure support IL projects/IL files?

I don't believe so.  @weshaggard, it doesn't, right?  I know that one or more of the corefxlab projects uses IL rewriting to support injecting IL, but I believe that's being done with some tools checked in to corefxlab rather than part of buildtools or corefx.

Comment (hughbe) [inline src/System.Reflection/tests/FieldInfoTests.cs]: Right - I can simulate these tests using Ref-Emit instead of IL, but then I'm uncertain as to whether I'm really testing Ref-Emit or everyday reflection.

Comment (weshaggard) [inline src/System.Reflection/tests/FieldInfoTests.cs]: We do have some basic support for ilproj's see example https://github.com/dotnet/corefx/blob/master/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj

Comment (stephentoub) [inline src/System.Reflection/tests/FieldInfoTests.cs]: Ah, excellent, I forgot we added the Unsafe project with IL support.  We only support IL-specific projects, right?   We don't support .csproj's that also include .il files?

Comment (weshaggard) [inline src/System.Reflection/tests/FieldInfoTests.cs]: Correct.
